examples/dataset/object_detection_3d/object_detection_3d/utils.py

```diff
@@ -97,3 +104,85 @@ def limit_period(val: np.array, offset: Optional[float] = 0.5, period: Optional[
         )
         for center, dim, yaw in zip(centers, dimensions, yaws)
     ]
+
+
+def kitti_matrix_to_array(arr: List[float], shape: Tuple[int, int]) -> np.array:
+    return np.array(arr).reshape(shape)
+
+
+def create_velo_to_pixel_matrix(velo_to_cam: List[float], cam_to_pixel: List[float]) -> List[List[float]]:
+    velo_to_cam_arr = kitti_matrix_to_array(velo_to_cam, (4, 4))
+    cam_to_pixel_arr = kitti_matrix_to_array(cam_to_pixel, (4, 4))
+    composed_arr = cam_to_pixel_arr @ velo_to_cam_arr  # type: ignore[operator]
+    return composed_arr[:3, :].tolist()
+
+
+def load_data(df_raw: pd.DataFrame) -> pd.DataFrame:
+    records = []
+    meta_cols = [col for col in df_raw.columns if col not in LABEL_FILE_COLUMNS]
+
+    for record in df_raw.itertuples():
+        bboxes_2d = [
+            LabeledBoundingBox(
+                label=box["label"],
+                top_left=tuple(box["left_top_left"]),
+                bottom_right=tuple(box["left_bottom_right"]),
+                occluded=box["occluded"],  # type: ignore[call-arg]
+                truncated=box["truncated"],  # type: ignore[call-arg]
+                alpha=box["alpha"],  # type: ignore[call-arg]
+                difficulty=box["difficulty"],  # type: ignore[call-arg]
+            )
+            for box in record.objects
+        ]
+        bboxes_3d = [
+            LabeledBoundingBox3D(
+                label=box["label"],
+                dimensions=(box["dim_x"], box["dim_y"], box["dim_z"]),
+                center=(box["loc_x"], box["loc_y"], box["loc_z"] + (box["dim_z"] / 2.0)),
+                # translate in z (up) axis to be in center
+                rotations=(0.0, 0.0, box["rotation_y"]),  # Z yaw axis for lidar coordinates
+                occluded=box["occluded"],  # type: ignore[call-arg]
+                truncated=box["truncated"],  # type: ignore[call-arg]
+                alpha=box["alpha"],  # type: ignore[call-arg]
+                difficulty=box["difficulty"],  # type: ignore[call-arg]
+            )
+            for box in record.objects
+        ]
+        counts = Counter([r["label"] for r in record.objects])
+        metadata = {col: getattr(record, col) for col in meta_cols}
+
+        records.append(
+            {
+                "image_id": record.image_id,
+                "locator": record.left_image,
+                "image_bboxes": bboxes_2d,
+                "images": [
+                    ImageAsset(
+                        locator=record.left_image,
+                        side="left",  # type: ignore[call-arg]
+                        # type: ignore[call-arg]
+                        projection=create_velo_to_pixel_matrix(record.Tr_velo_to_cam, record.P2),
+                        velodyne_bboxes=bboxes_3d,  # type: ignore[call-arg]
```

Comment:
Do we need to attach the 3d bboxes to the image assets? I thought they'd come from the parent level?